src/sp_utils.c

```diff
@@ -277,63 +268,50 @@ void sp_log_disable_ret(const char* restrict path,
   }
 }
 
-int sp_match_array_key(const zval* zv, const char* to_match, const pcre* rx) {
+bool sp_match_array_key(const zval* zv, const char* to_match, const pcre* rx) {
   zend_string* key;
-  zval* value;
-  char* arg_value_str;
+  zend_ulong idx;
 
-  ZEND_HASH_FOREACH_STR_KEY_VAL(Z_ARRVAL_P(zv), key, value) {
-    if (Z_TYPE_P(value) == IS_ARRAY) {
-      continue;
-    }
-    arg_value_str = sp_convert_to_string(value);
-    if (!sp_match_value(arg_value_str, to_match, rx)) {
-      efree(arg_value_str);
-      continue;
+  ZEND_HASH_FOREACH_KEY(Z_ARRVAL_P(zv), idx, key) {
+    if (key) {
+      if (sp_match_value(ZSTR_VAL(key), to_match, rx)) {
+	return true;
+      }
     } else {
-      efree(arg_value_str);
-      return 1;
+      char *idx_str = NULL;
+
+      // Could use a log.
+      idx_str = emalloc(snprintf(NULL, 0, "%lu", idx));
```

Comment:
snprintf doesn't count the null byte at the end, maybe checking as well it's no a negative return value ?